### PR TITLE
MGDOBR-985: add error handling cypress tests

### PR DESCRIPTION
**Pull request by jomarko**
Dependent on #145

### CODE REVIEW

File: cypress/integration/InstancesTest.ts [hunk 1]
```diff
@@ -272,6 +272,41 @@ describe("Instances Test", () => {
...
+        // visit 'Instance one'
+        cy.visit("/instance/3543edaa-1851-4ad7-96be-ebde7d20d717");
+
+        cy.ouiaId("error-handling", "PF4/TabButton").click();
+
+        cy.ouiaId("error-handling-section", "PF4/Text")
+          .should("contain.text", "Error handling method")
+          .should("be.visible");
+
+        cy.get(".pf-c-description-list__term")
```
[the remark this answers is not in the document]
Comment (sjamboro) [reply cypress/integration/InstancesTest.ts:286]: I was able to add ouiaId to each component. See:
image

There are the desired elements/components:
https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/blob/main/src/app/Processor/ProcessorConfigParameters/ProcessorConfigParameters.tsx#L29
https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/blob/main/src/app/Processor/ProcessorConfigParameters/ProcessorConfigParameters.tsx#L50

File: cypress/integration/InstancesTest.ts [hunk 2]
```diff
@@ -272,6 +272,41 @@ describe("Instances Test", () => {
...
+          .should("have.length.gt", 1)
+          .should(($items) => {
+            expect($items).to.have.length(5);
```
[the remark this answers is not in the document]
Comment (jomarko) [reply cypress/integration/InstancesTest.ts:289]: @sjamboro yes, thank you for a question, during the test creation, I have an impression that there is some delay between opening the details panel, and showing all attributes. that is why I constructed the check in this way.

Please tell me what is a desired state form you point of view:
- remove `.should("have.length.gt", 1)` check
- replace `.should("have.length.gt", 1)` check with `.should("have.length", 5)`
- replace `.should("have.length.gt", 1)` check with ...
- ...

?

File: cypress/integration/InstancesTest.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jomarko) [reply cypress/integration/InstancesTest.ts:286]: I will try, sure, thank you

File: cypress/integration/InstancesTest.ts [hunk 3]
```diff
@@ -272,6 +272,41 @@ describe("Instances Test", () => {
...
+            expect($items.eq(0)).to.contain.text("Error handling method");
```
[the remark this answers is not in the document]
Comment (jomarko) [reply cypress/integration/InstancesTest.ts:290]: @sjamboro just personal preference, to make assertion safe from changes like:
- "Error handling method:"
- "Error handling method -"
- "[icon] Error handling method"
- ...

I can use `have`if you wish

**Comment by jomarko** [comment]
@sjamboro please re-review #169. I believe I addressed most your comments. Some of my changes may change your opinion about some point I didn't address. Please let me know what are your thoughts about the PR now.

File: cypress/integration/InstancesTest.ts [hunk 4]
```diff
@@ -272,6 +272,82 @@ describe("Instances Test", () => {
...
+        cy.ouiaId("basic_auth_password", "ProcessorConfig/FormGroup").within(
+          () => {
+            cy.get("[data-testid='basic_auth_password']").should(
+              "contain.text",
+              "Basic Auth Password"
+            );
+            cy.get("[data-testid='basic_auth_password-value']").should(
+              "contain.text",
+              "****"
+            );
```
Comment (ChughShilpa) [inline cypress/integration/InstancesTest.ts:331]: IMO, Password must be asserted with 'have' as this value is constant

Comment (ChughShilpa) [reply cypress/integration/InstancesTest.ts:331]: Same opinion for other fields as well.

Comment (jomarko) [reply cypress/integration/InstancesTest.ts:331]: pushing in a separate commit, thank you

File: cypress/integration/InstancesTest.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (sjamboro) [reply cypress/integration/InstancesTest.ts]: @jomarko yes I understand but my personal preferences are opposite :D (I would like to know that the label is changed)

**Review by sjamboro — APPROVED** [review]
Thank you ... It looks good.

Please check that all tests are performed in GHA (few minutes ago there was performed only Instance Error Handler test).

**Review by ChughShilpa — APPROVED** [review]
Thanks for the update

**Comment by sjamboro** [comment]
> @jomarko LGTM. thanks. Few observation/queries not from the PR Do we have test coverage for instance creation with error handler type and also noticed that for Error Handling in UI we don't have option to edit the parameters but our API supports

I do not think so. Error Handling is quite new in UI.

**Comment by jomarko** [comment]
@sjamboro yes, now tests are executed. It had pushed `it.only` by mistake. Fixed by force push.

@Srihari1192 about the edit error handler configuration update. It is opened right now. I added you as reviewer https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/204

@Srihari1192 creation of RHOSE instance with error handling strategy is not covered. Please create a jira and feel free to contribute with such  test. Please be aware of realted test that is pending from me https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/203

**Comment by Srihari1192** [comment]
> @sjamboro yes, now tests are executed. It had pushed `it.only` by mistake. Fixed by force push.
> 
> @Srihari1192 about the edit error handler configuration update. It is opened right now. I added you as reviewer #204
> 
> @Srihari1192 creation of RHOSE instance with error handling strategy is not covered. Please create a jira and feel free to contribute with such test. Please be aware of realted test that is pending from me #203

@jomarko Sure thanks for the info

**Pull request merged**